lib/galaxy_test/api/test_tools.py

```diff
@@ -1589,14 +1590,39 @@ def test_identifier_map_over_multiple_input_in_conditional(self, history_id):
         output1_content = self.dataset_populator.get_history_dataset_content(history_id, dataset=output1)
         self.assertEqual(output1_content.strip(), "forward\nreverse")
 
+    @skip_without_tool("identifier_in_conditional")
+    @uses_test_history(require_new=False)
+    def test_identifier_map_over_multiple_input_in_conditional_new_payload_form(self, history_id):
+        hdca_id = self._build_pair(history_id, ["123", "456"])
+        inputs = {
+            "outer_cond": {
+                'multi_input': True,
+                '__current_case__': 0,
```

Comment:
Is this necessary for the tests to pass ?

Reply:
Yes, but I'm not sure why.

Reply:
😿  I was hoping we'd get a nicer input format out of this. I'll have a look if we can do something about this.

Reply:
To be a bit more helpful: leaving them out causes a KeyError here https://github.com/galaxyproject/galaxy/blob/ca442596c344742c2a7f64d8d0bedac41536c88b/lib/galaxy/tools/parameters/__init__.py#L245

and here

https://github.com/galaxyproject/galaxy/blob/ca442596c344742c2a7f64d8d0bedac41536c88b/lib/galaxy/tools/parameters/__init__.py#L247

Reply:
Ideally we wouldn't need to create `incoming`, the dictionary should map cleanly onto parameters, so what I would try is replacing populate_state (https://github.com/mvdbeek/galaxy/blob/dev/lib/galaxy/tools/parameters/__init__.py#L274) with a simpler version that works with a clean dictionary. That's pretty deep in Galaxy's internals though, and I'm not sure it would actually work ... but if it did that would be amazing.
Let me know if you're up for giving this a try, otherwise I'm happy to take a look.